projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts

```diff
@@ -6460,3 +6494,85 @@ const verifyChipVisibility = (fix, index: number, shouldBeFullyVisible: boolean)
     expect(chipRect.left >= visibleChipAreaRect.left && chipRect.right <= visibleChipAreaRect.right)
         .toBe(shouldBeFullyVisible, 'chip[' + index + '] visibility is incorrect');
 };
+
+const emitFilteringDoneOnResetClick = (fix, grid, filterVal: any, columnName: string, condition) => {
+    filterGrid(fix, grid, columnName, filterVal, condition);
+
+    const filteringExpressions = grid.filteringExpressionsTree.find(columnName) as FilteringExpressionsTree;
+    verifyEmitFilteringDone(grid, filteringExpressions, 1);
+
+    GridFunctions.clickFilterCellChip(fix, columnName);
+    GridFunctions.resetFilterRow(fix);
+
+    const emptyFilter = new FilteringExpressionsTree(null, columnName);
+    verifyEmitFilteringDone(grid, emptyFilter, 2);
+
+    const filterUiRow = fix.debugElement.query(By.css(FILTER_UI_ROW));
+    const reset = filterUiRow.queryAll(By.css('button'))[0];
+    expect(reset.nativeElement.classList.contains('igx-button--disabled')).toEqual(true);
+};
+
+const emitFilteringDoneOnInputClear = (fix, grid, filterVal, columnName, condition) => {
+    filterGrid(fix, grid, columnName, filterVal, condition);
+
+    GridFunctions.clickFilterCellChip(fix, columnName);
+
+    spyOn(grid.filteringDone, 'emit');
```

Comment:
Isn't this spy a duplicate of the one in beforeEach?

Reply:
Actually yes, it is duplicated. So I will remove it.